pkg/config/get.go

```diff
@@ -0,0 +1,40 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func pathSplit(r rune) bool {
+	return r == '.' || r == '[' || r == ']' || r == '"'
+}
+
+func GetValueFromConfig(stringPath string, object interface{}) (interface{}, error) {
```

Comment:
This looks good to me but can we have a few unit tests for this function? A good case for ["table driven testing"](https://github.com/golang/go/wiki/TableDrivenTests).

- test for a simple key in cluster
- test for a path with `[]` and `.`

Reply:
Although, we should use the `assert` package, like: https://github.com/jkcfg/jk/blob/master/pkg/resolve/std_importer_test.go

Reply:
Ok, I will do that.
the '"' is a special key added for the EndToEnd tests which keep quotes that are required in some cases like `footloose config get "machines[0].spec"`

Reply:
Ah, that's because the end to end test don't need the `""`.

The reason why you need them on the command line is because the shell will interpret the `[` and `]` so you need the quotes to protect those characters.

The tests commands don't (currently) run into a shell, they are directly given to `exec` so we don't need the quotes there.

Reply:
I have just tested, indeed I can remove the quotes in the test file :)